.clang-tidy

```diff
@@ -84,6 +90,16 @@ hicpp-*,
 #        void bar(Foo& foo);
 #        // in cpp file
 #        void bar(Foo& foo) {...}
+#
+# * rule: cppcoreguidelines-non-private-member-variables-in-classes
+#   justification: Sometimes it makes sense to have protected members to extend the base class.
+#                  Even public members are useful in lightweight classes (effectively structs).
+#                  This unnecessarily limits design options.
+#
+# * rule: cppcoreguidelines-avoid-c-arrays
+# * rule: hicpp-avoid-c-arrays
+#   justicfication: C arrays are useful in low level code for buffers etc.
```

Comment:
@elBoberido If we do not want to suppress this globally I will add a suppression comment where it applies.

Note that I think using C arrays should be allowed and would be in favor of suppression.